File: 001_twoSum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        
        var dict = new Dictionary<int, int>();
        for (int i=0; i<nums.Length; i++) {
            if (dict.ContainsKey(target-nums[i]))
				// return {dict[target-nums[i]], i} directly causes syntax error
                return new int[]{dict[target-nums[i]], i};
            dict[nums[i]] = i;
        }
        return new int[2];
    }
}

File: 001_twoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
		// use default initializer
        vector<int> result;
        // use default initializer
		unordered_map<int, int> dict;
        for (int i = 0;i<nums.size(); i++) {
            if (dict.find(target-nums[i]) != dict.end()) {
                result.push_back(dict[target - nums[i]]);
                result.push_back(i);
                break;
            }
            dict[nums[i]] = i;
        }
        return result;
    }
};